### PR TITLE
Port TGUI react migration and fixes, Part 1

### PR DESCRIPTION
**Pull request by Doubleumc**
# About the pull request

Ports various commits related to main CM's upgrade of TGUI:
- https://github.com/cmss13-devs/cmss13/pull/5307
- https://github.com/cmss13-devs/cmss13/pull/5435
- https://github.com/cmss13-devs/cmss13/pull/5493
- https://github.com/cmss13-devs/cmss13/pull/5562
- https://github.com/cmss13-devs/cmss13/pull/5581
- https://github.com/cmss13-devs/cmss13/pull/5649
- https://github.com/cmss13-devs/cmss13/pull/5651
- https://github.com/cmss13-devs/cmss13/pull/5764
- https://github.com/cmss13-devs/cmss13/pull/5775
- https://github.com/cmss13-devs/cmss13/pull/6196
- https://github.com/cmss13-devs/cmss13/pull/6244

Additionally several misc commits that TGUI upgrades depended on:
- https://github.com/cmss13-devs/cmss13/pull/5584
- https://github.com/cmss13-devs/cmss13/pull/5158
- https://github.com/cmss13-devs/cmss13/pull/5165
- https://github.com/cmss13-devs/cmss13/pull/5417

It boots, nothing looks obviously wrong. Beyond that, will need some serious testing. Changes that rely on changes that rely on changes that rely on changes -- I traced them back as best I could, but there's always a chance that some code change required to make these changes work got missed.

Assuming that all goes well, would check for any differences in TGUI files to ensure everything up-to-date with PvP.

# Explain why it's good for the game

React is a much better supported framework than Inferno. Modern TGUI has performance and functionality improvements.

# Testing Photographs and Procedure

Its too big. Gonna have to be testmerged.

# Changelog

:cl:
refactor: complete overhaul of just about every TGUI interface
add: added parachutes
add: added holocards to the squad hud
add: falling out a dropship throws you to the surface
/:cl:

### CODE REVIEW

File: code/modules/shuttle/shuttles/trijent_elevator.dm [hunk 1]
```diff
@@ -27,6 +27,8 @@
 	door_control.label = "elevator"
 	for(var/area/shuttle_area in shuttle_areas)
 		for(var/obj/structure/machinery/door/door in shuttle_area)
+			if(isnum(door.id)) //poddoor filler was sneaking in
```
[the remark this answers is not in the document]
Comment (Doubleumc) [reply code/modules/shuttle/shuttles/trijent_elevator.dm:30]: Sure thing. My thinking was the real issue is it uses the `door.id` as the index for an assoc list, and a number index really throws a wrench in that. Regardless, it fixes the problem.

**Review by Drulikar — APPROVED** [review]
Of note, a bulk of this is my own code so take my approval with a grain of salt. Feel free to let me know if you find anything that needs tweaking though. I will port most other tgui changes I think will be useful here once this is merged so don't be scared with a big PR influx.

**Review by Drulikar — CHANGES_REQUESTED** [review]
Dropship equipment is not rendering correctly and as such is not interactable. Paradrop equipment for example would need to be accessed here to be used, but it cannot because the mount point is null.

image
image
image

**Comment by Doubleumc** [comment]
Good catch. Tested and fixed for all PvE dropships.

**Review by Drulikar — CHANGES_REQUESTED** [review]
All hardpoints look fixed and cas firing is fine.

Paradropping does not transition the person to the map however. It seems `/turf/open/space/transit/proc/handle_crosser(atom/movable/crosser)` is triggered but not `/turf/open/space/transit/dropship/handle_crosser(atom/movable/crosser)` so maybe the shuttles need further mapping changes to be more like pvp?

para

**Comment by Doubleumc** [comment]
Good catch, and fixed!
image

**Review by Drulikar — APPROVED** [review]
Let me know if anything else comes up and or you feel its been tested enough.

**Comment by Doubleumc** [comment]
One full mission without any issues being noticed and I'll be satisfied.

**Comment by Doubleumc** [comment]
I'm satisfied.

**Pull request merged**